core/cog-launcher.c

```diff
@@ -22,6 +22,7 @@
 struct _CogLauncher {
     CogLauncherBase parent;
     CogShell       *shell;
+    gchar          *permissions;
```

Comment:
I would use a boolean or enum value here, instead of keeping the string around.

Reply:
Long term we will need a proper data structure here.

Reply:
That's absolutely right, bit for now we're keeping it as simple as possible.

Reply:
Yeah... and a string wasn't a proper data structure anyway, you were totally right ;)